crates/bevy_transform/src/components/global_transform.rs

```diff
@@ -25,223 +24,87 @@ use std::ops::Mul;
-        self.look_at(target, up);
-        self
-    }
-
-    #[doc(hidden)]
-    #[inline]
-    #[must_use]
-    pub const fn with_translation(mut self, translation: Vec3) -> Self {
-        self.translation = translation;
-        self
-    }
-
-    #[doc(hidden)]
-    #[inline]
-    #[must_use]
-    pub const fn with_rotation(mut self, rotation: Quat) -> Self {
-        self.rotation = rotation;
-        self
-    }
-
-    #[doc(hidden)]
-    #[inline]
-    #[must_use]
-    pub const fn with_scale(mut self, scale: Vec3) -> Self {
-        self.scale = scale;
-        self
-    }
-
-    /// Returns the 3d affine transformation matrix from this transforms translation,
-    /// rotation, and scale.
-    #[inline]
-    pub fn compute_matrix(&self) -> Mat4 {
-        Mat4::from_scale_rotation_translation(self.scale, self.rotation, self.translation)
-    }
-
-    /// Returns the 3d affine transformation from this transforms translation,
-    /// rotation, and scale.
+    /// Creates a new identity [`GlobalTransform`], that maps all points in space to themselves.
     #[inline]
-    pub fn compute_affine(&self) -> Affine3A {
-        Affine3A::from_scale_rotation_translation(self.scale, self.rotation, self.translation)
-    }
-
-    /// Get the unit vector in the local x direction
-    #[inline]
-    pub fn local_x(&self) -> Vec3 {
-        self.rotation * Vec3::X
-    }
-
-    /// Equivalent to [`-local_x()`][GlobalTransform::local_x]
-    #[inline]
-    pub fn left(&self) -> Vec3 {
-        -self.local_x()
-    }
-
-    /// Equivalent to [`local_x()`][GlobalTransform::local_x]
-    #[inline]
-    pub fn right(&self) -> Vec3 {
-        self.local_x()
-    }
-
-    /// Get the unit vector in the local y direction
-    #[inline]
-    pub fn local_y(&self) -> Vec3 {
-        self.rotation * Vec3::Y
-    }
-
-    /// Equivalent to [`local_y()`][GlobalTransform::local_y]
-    #[inline]
-    pub fn up(&self) -> Vec3 {
-        self.local_y()
-    }
-
-    /// Equivalent to [`-local_y()`][GlobalTransform::local_y]
-    #[inline]
-    pub fn down(&self) -> Vec3 {
-        -self.local_y()
+    pub const fn identity() -> Self {
+        Self(Affine3A::IDENTITY)
     }
 
-    /// Get the unit vector in the local z direction
-    #[inline]
-    pub fn local_z(&self) -> Vec3 {
-        self.rotation * Vec3::Z
-    }
+    impl_local_axis!(right, left, X);
+    impl_local_axis!(up, down, Y);
+    impl_local_axis!(back, forward, Z);
 
-    /// Equivalent to [`-local_z()`][GlobalTransform::local_z]
+    /// Get the translation as a [`Vec3`].
     #[inline]
-    pub fn forward(&self) -> Vec3 {
-        -self.local_z()
+    pub fn translation(&self) -> Vec3 {
+        self.0.translation.into()
     }
 
-    /// Equivalent to [`local_z()`][GlobalTransform::local_z]
+    /// Get the translation as a [`Vec3A`].
     #[inline]
-    pub fn back(&self) -> Vec3 {
-        self.local_z()
+    pub fn translation_vec3a(&self) -> Vec3A {
```

Comment:
I think this approach in general makes sense for dealing with Vec3 vs Vec3A, and it doesn't break existing API which is nice too.